### ISSUE TITLE
scipy.optimize.minimize automatically convert scalar x0 to list

### ISSUE THREAD

**Issue by nosarthur** [scipy/scipy#13743]
My issue is about `scipy.optimize.minimize`. When the input `x0` is a scalar, it is converted to a list. Since the user defined object function still assumes `x0` to be a scalar, it can cause `VisibleDeprecationWarning`. I think a solution could be simply raising a traceback for scalar input  

#### Reproducing code example:

```
from scipy.optimize import minimize
import numpy as np
def g(x):
    print(x)
    y = np.array([1, x, 3])
    return x * x
minimize(g, 2, method='BFGS')
```

#### Error message:

```
[2.]
<ipython-input-18-7dbf1cec2b87>:5: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray
  y = np.array([1, x, 3])
[2.00000001]
[0.99]
[0.99000001]
[-1.88846401e-08]
[-3.98347888e-09]
```

#### Scipy/Numpy/Python version information:

```
serial 1.5.3 1.19.4 sys.version_info(major=3, minor=8, micro=3, releaselevel='final', serial=0)
```

**Comment by ev-br** [comment]
My knee-jerk reaction is that `np.array([1, x, 3])` is a user error, it should probably be `np.r_[1, x, 3]` or some such.

**Comment by mdhaber** [comment]
I agree. The documentation states that `x` will be passed in as a 1D array. I don't think we should raise an error when `x0` is a scalar because that would raise errors in existing code. Does that make sense @nosarthur ?

**Comment by nosarthur** [comment]
I agree it is a user error. In my real code, the object function is related to `y` and it actually behaves as expected (for example, you can change the return to `np.linalg.norm(y)`) Maybe `minimize` first flattens `x0`.  It is thus not noticed until this warning shows up. 

Not raising error sounds good too. I guess at some point this warning will become an error anyways. It is just slightly less obvious what the issue is.

**Comment by mdhaber** [comment]
OK. If we can't raise an error, are there changes to the documentation that would help?

**Comment by nosarthur** [comment]
The document of `scipy.optimize.minimize` is quite clear. Please feel free to close the case. Thanks
